### PR TITLE
FIX(server): Packet size issues

### PR DESCRIPTION
**Pull request by Krzmbrzl**
This PR fixes 2 issues in the server's code that have something to do with packet sizes:
1. Make sure `poslen <= len` in order for `len - poslen` to never get negative (which can cause trouble down the line)
2. Enforce the maximum package size for packets received via UDPTunnel as well

### CODE REVIEW

File: src/murmur/Server.cpp [hunk 1]
```diff
@@ -1110,6 +1110,14 @@ void Server::processMsg(ServerUser *u, const char *data, int len) {
 
 	len = pds.size() + 1;
 
+	if (poslen > static_cast<unsigned int>(len)) {
```
Comment (davidebeatrici) [inline src/murmur/Server.cpp:1113]: I assume the cast is needed due to the compiler complaining about unsigned & signed comparison, right?

Hopefully once we switch to POCO we don't have to deal with such cases anymore...

Comment (Krzmbrzl) [reply src/murmur/Server.cpp:1113]: Yes exactly. How would POCO solve this issue though. signed and unsigned integers are different things. POCO doesn't change that :thinking:

Comment (davidebeatrici) [reply src/murmur/Server.cpp:1113]: I'm pretty sure POCO uses `size_t`.

Comment (Krzmbrzl) [reply src/murmur/Server.cpp:1113]: Ah okay - yeah that'd probably solve the issue here :)

**Pull request merged**